### ISSUE TITLE
Faster `Tile.prototype._createTextureData`

### ISSUE THREAD

**Issue by caewok** [foundryvtt/foundryvtt#8827]
### User Experience

I think `Tile.prototype._createTextureData` could be made to be a bit faster and more space-efficient.

First, it doesn't seem necessary to construct a sprite nor render to a texture in this portion of the code:
```js
    // Else, we are preparing the texture data creation
    const map = this._textureData;

    // Create a temporary Sprite using the Tile texture
    const sprite = new PIXI.Sprite(this.texture);
    sprite.width = map.aw = this.texture.baseTexture.realWidth / 4;
    sprite.height = map.ah = this.texture.baseTexture.realHeight / 4;
    sprite.anchor.set(0.5, 0.5);
    sprite.position.set(map.aw / 2, map.ah / 2);

    // Create or update the alphaMap render texture
    const tex = PIXI.RenderTexture.create({width: map.aw, height: map.ah});

    // Render the sprite to the texture and extract its pixels
    // Destroy sprite and texture when they are no longer needed
    canvas.app.renderer.render(sprite, tex);
    sprite.destroy(false);
    const pixels = map.pixels = canvas.app.renderer.extract.pixels(tex);
    tex.destroy(true);
```

Instead, just use the tile texture directly, and then filter the resulting pixel array manually if you want to lower the resolution to save space and time:
```js
    // Else, we are preparing the texture data creation
    const map = this._textureData;
    
    const { pixels, width, height } = canvas.app.renderer.extract.pixels(this.texture);    
    
    // Shrink by 75%
    map.pixels = pixels.filter((_px, i) => (i % 16) < 4);
    map.aw = width / 4;
    map.ah = height / 4;
```

This latter code performs ~ 15% faster in my test, and appears to be equivalent output of pixels (I checked the pixel sums, which remain the same). There are probably times when it makes sense to downsample to a sprite if you are doing fancy antialiasing or something, but that does not appear to be the case here.

Second, instead of copying the alpha channel to the red channel and storing all 4 (a, b, g, a) channels, you could use that same filter to get only alpha channel, saving 75% of the space:

```js
map.pixels = pixels.filter((_px, i) => (i % 16) === 3);
```

Third, `Tile.prototype.#getPixelAlpha` rounds the `aw` width property: 
```js
    // Next test a specific pixel
    const px = (Math.floor(y) * Math.roundFast(Math.abs(this._textureData.aw))) + Math.floor(x);
```

Instead, why not just round it when setting it in the first place in `Tile.prototype._createTextureData`? (Also, I don't see how `aw` could be negative, but assuming it is, you could take the absolute value there as well.) So:

```js
    map.aw = Math.roundFast(Math.abs(width / 4));
    map.ah =  Math.roundFast(Math.abs(height / 4));
```

**Comment by caewok** [comment]
Addendum:

Something is different about `canvas.app.renderer.extract.pixels` versus the extract function I borrowed from Perfect Vision. 
https://github.com/caewok/fvtt-elevated-vision/blob/master/scripts/perfect-vision/extract-pixels.js#L10

`canvas.app.renderer.extract.pixels(this.texture)` is throwing an error: "TypeError: t.getLocalBounds is not a function" in Foundry v10. This might be different for newer versions of PIXI in Foundry v11. Using `extractPixels(canvas.app.renderer, this.texture)`, on the other hand, works fine. 

I know I had `canvas.app.renderer.extract.pixels(this.texture)` working previously, because I had benchmarked it against `extractPixels.` So I don't know what is now causing this error, given I did not see it previously. Maybe some tiles work but some don't using `canvas.app.renderer.extract.pixels(this.texture)`?

(FYI, Perfect Vision is now using a more complicated worker to do pixel extraction. https://github.com/dev7355608/perfect-vision/blob/main/scripts/utils/extract.js)

Bottom line, if you want to switch to the faster approach of using `tile.texture`, make sure to test with a variety of tiles and possibly avoid `canvas.app.renderer.extract.pixels` if you cannot address the error.

**Comment by caewok** [comment]
Nevermind; I figured out the issue. `_createTextureData` was pulling back in the texture data from the `canvas.tiles.textureDataMap`. So the benchmark was not valid. I am closing this as I don't think the approach above will work, given that. I will open a new issue if I can find a better approach.

**Comment by aaclayton** [comment]
Thanks for investigating this, evne if it didn't result in the improvement you had hoped.
